sandy/core/cov.py

```diff
@@ -951,10 +984,33 @@ def corr2cov(corr, s):
     -------
     `numpy.ndarray`
         covariance matrix
+
+    Examples
+    --------
+    >>> S = np.array([1,2,3])
+    >>> var = np.array([[1, 0, 2, 0], [0, 3, 0, 0], [4, 0, 5, 0], [0, 0, 0, 0]])
+    >>> corr2cov(var,S)
+    array([[ 1,  0,  6,  0],
+           [ 0, 12,  0,  0],
+           [12,  0, 45,  0],
+           [ 0,  0,  0,  0]])
+
+    >>> S = np.array([1,2,3])
+    >>> var = np.array([[1, 0, 2], [0, 3, 0], [4, 0, 5]])
+    >>> corr2cov(var,S)
+    array([[ 1,  0,  6],
+           [ 0, 12,  0],
+           [12,  0, 45]])
     """
     dim = corr.shape[0]
-    S = np.repeat(s, dim).reshape(dim, dim)
-    return S.T * (corr * S)
+    if len(s) > dim:
+        raise sandy.Error("The shape of the variables is not correct")

```

Comment:
I would like to stop using `sandy.Error`...maybe use `TypeError`?
you can find [here](https://docs.python.org/3/library/exceptions.html#bltin-exceptions) a list of errors that you can use

Reply:
please, fix this